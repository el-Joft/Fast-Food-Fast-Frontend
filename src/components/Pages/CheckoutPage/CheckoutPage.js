/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import './CheckoutPage.scss';
import { changeQuantity, checkoutOrder } from '../../../actions/orderAction';


class CheckoutPage extends Component {
  submitData = (e) => {
    e.preventDefault();
    const { isAuthenticated, user } = this.props.auth;
    const { history } = this.props;
    if (isAuthenticated) {
      const submitValue = this.props.cart.map(item => ({ menuid: item.id, quantity: item.quantity, orderedby: user.id }));
      let value = null;
      const token = localStorage.getItem('userToken');
      if (submitValue.length > 1) {
        submitValue.map((data) => {
          value = {
            menuid: data.menuid,
            quantity: data.quantity,
            orderedby: user.id,
          };
          this.props.checkoutOrder(value, token, history);
        });
      } else {
        this.props.checkoutOrder(value, token, history);
        console.log('solutions', token);
      }
    }
  };

  changeInput = (e, id) => {
    const value = Number(e.target.value);
    this.props.changeQuantity(id, value);
  };

  renderCart = () => this.props.cart.map(item => (
    <div key={item.id} className="checkout-header">
      <div className="checkout-item top-header">
        <h3>Item</h3>
        <p>
          {item.name}
          <input type="text" name="menuid" value={item.name} hidden />
        </p>
      </div>
      <div className="top-header checkout-quantity">
        <h3>Quantity</h3>
        <input
          type="number"
          min="1"
          max="100"
          value={item.quantity}
          onChange={e => this.changeInput(e, item.id)}
          onKeyDown={this.getTotal}
          className="checkout-input"
        />
      </div>
      <div className="top-header checkout-price">
        <h3>Price</h3>
        <p id="{item.price}">
          <strike>N</strike>
          {item.price}
        </p>
      </div>
      <div className="top-header checkout-subtotal">
        <h3>Sub-Total</h3>
        <p>
          <strike>N</strike>
          { item.quantity * item.price }
        </p>
      </div>
    </div>

  ));

  render() {
    const { isAuthenticated } = this.props.auth;
    const { cart } = this.props;
    return (
      <div>
        { cart
          ? (
            <div className="checkout">
              <form onSubmit={e => this.submitData(e)}>
                {this.renderCart()}
                <div className="top-button">
                  <button type="submit" className="cancel-order">
            Cancel Order
                    {' '}
                  </button>
                  { isAuthenticated
                    ? (
                      <button type="submit" className="complete-order ">
            Complete Order

                      </button>
                    )

                    : <button type="submit" className="complete-order" disabled> Login to continue</button>
              }
                </div>
              </form>
            </div>
          )
          : (
            <div className='cart-empty'>
              <div>Your cart is currently empty</div>
              <a href='/menu'>go the menu ....</a>
            </div>
          )
        }
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  cart: state.order.cart,
});

export default connect(mapStateToProps, { changeQuantity, checkoutOrder })(CheckoutPage);
