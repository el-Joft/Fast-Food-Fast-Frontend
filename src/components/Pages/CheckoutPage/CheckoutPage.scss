
.checkout-header{
  display: flex;
  
  /* width: 80%; */
 
}
.more-order{
  border: 2px solid #42ce9f;
  padding: 10px;
  width: 200px;
  text-align: center;
  margin-bottom: 20px;
}
.top-header{
  padding: 5px;
}
.top-header p{
  margin-top: 12px;
  
}
.checkout{
  margin-top: 20px;
  width: 70%;
  margin-bottom: 50px;
 
}
.checkout-item{
  flex-grow: 5;
  width: 60%;
  
}
.checkout-input{
  border: none;
  border-bottom: 2px solid #42ce9f;
  width: 60%;
  text-align: center;
}

.checkout-quantity{
  flex-grow: 1;
}
.checkout-price{
  flex-grow: 2;
}
.checkout-subtotal{
  flex-grow: 3;
}
.top-body{
  text-align: center; 
}
.top-button{
  width: 30%;
  text-align: center;
  display: flex;
  margin-bottom: 30px;
  margin-top: 30px;
}
.cancel-order {
  color: #ff0a10;
  padding: 10px;
  width: 45%;
  border: 2px solid #ff0a10;
  border-radius: 10%;
}
.complete-order {
  color: #42ce9f;
  padding: 10px;
  width: 45%!important;
  border: 2px solid #42ce9f;
  border-radius: 10%;
}
table,thead,tbody,tfoot,tr, td,th{
  text-align: center;
  margin: auto;
  border:1px solid #dedede;
  padding: 1rem;
  /* width: 50%; */
}
.menu_links {
  cursor: pointer;
}

.successful{
  text-align: center;

}
.successful img{
  width: 40%;
}
.successful h3{
  font-size: 1.4em;
}
.cart-empty {
  margin: 10px;
  text-align: center;
  padding: 50px;
  font-size: 2em;
}