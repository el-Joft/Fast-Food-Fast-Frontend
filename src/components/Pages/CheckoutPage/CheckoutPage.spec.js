/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { CheckoutPage } from './CheckoutPage';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      changeQuantity: jest.fn(),
      checkoutOrder: jest.fn(),
      history: '',
      success: {
        status: true,
      },
      orders: [],
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
        },
      },
    };
  });

  it('invokes `componentDidMount` when mounted', () => {
    const spy = jest.spyOn(CheckoutPage.prototype, 'componentDidMount');
    wrapper = shallow(<CheckoutPage {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render the component', () => {
    wrapper = shallow(<CheckoutPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Button component', () => {
    wrapper = shallow(<CheckoutPage {...props} />);
    expect(wrapper.length).toBe(1);
  });
});
