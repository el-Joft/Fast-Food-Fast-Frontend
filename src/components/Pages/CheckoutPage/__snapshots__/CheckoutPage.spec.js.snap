// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPage
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "id": 1,
        },
      }
    }
    changeQuantity={[MockFunction]}
    checkoutOrder={[MockFunction]}
    history=""
    orders={Array []}
    success={
      Object {
        "status": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cart-empty"
      >
        <div>
          Your cart is currently empty
        </div>
        <a
          href="/menu"
        >
          go the menu ....
        </a>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Your cart is currently empty
          </div>,
          <a
            href="/menu"
          >
            go the menu ....
          </a>,
        ],
        "className": "cart-empty",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your cart is currently empty",
          },
          "ref": null,
          "rendered": "Your cart is currently empty",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "go the menu ....",
            "href": "/menu",
          },
          "ref": null,
          "rendered": "go the menu ....",
          "type": "a",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cart-empty"
        >
          <div>
            Your cart is currently empty
          </div>
          <a
            href="/menu"
          >
            go the menu ....
          </a>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Your cart is currently empty
            </div>,
            <a
              href="/menu"
            >
              go the menu ....
            </a>,
          ],
          "className": "cart-empty",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your cart is currently empty",
            },
            "ref": null,
            "rendered": "Your cart is currently empty",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "go the menu ....",
              "href": "/menu",
            },
            "ref": null,
            "rendered": "go the menu ....",
            "type": "a",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
